include/alpaka/dev/common/QueueRegistry.hpp

```diff
@@ -5,14 +5,21 @@
 #pragma once
 
 #include "alpaka/core/Common.hpp"
+#include "alpaka/dev/common/DeviceProperties.hpp"
 
 #include <deque>
 #include <functional>
 #include <memory>
 #include <mutex>
+#include <optional>
 
 namespace alpaka::detail
 {
+
+    inline void setDeviceProperties(int iDevice, alpaka::DeviceProperties& devProperties);
+    template<typename TApi>
+    inline void setDeviceProperties(int iDevice, alpaka::DeviceProperties& devProperties);
```

Comment:
I wonder if this could be declared as
```
    template<typename TDev>
    inline void setDeviceProperties(TDev const& device, alpaka::DeviceProperties& devProperties);
```
and then specialised for the `DevCpu` and `DevUniformCudaHipRt<TApi>` cases ?

That is, passing the device as an argument instead of the index.